spring-security-karaf-login-module/src/main/java/org/pentaho/platform/osgi/auth/spring/SpringSecurityLoginModule.java

```diff
@@ -181,13 +182,30 @@ public boolean login() throws LoginException {
     // If they have AdministerSecurity, grant the Karaf admin role
     if ( getAuthorizationPolicy().isAllowed( AdministerSecurityAction.NAME ) ) {
       principals.add( new RolePrincipal( KARAF_ADMIN ) );
```

Comment:
Not sure the role `"karaf_admin"` is used anywhere outside of this class

- github search yields no results : https://github.com/search?q=org%3Apentaho+karaf_admin&type=code
- did local grep of 9.6 PBA installation with no results: https://hv-eng.atlassian.net/browse/BACKLOG-37812?focusedCommentId=1771630

Reply:
The `SpringSecurityLoginModule.KARAF_ADMIN` constant has the same value of the property `karaf.admin.role` we used to have in `assemblies/server/src/main/resources-filtered/etc/system.properties` (removed in https://github.com/pentaho/pentaho-karaf-ee-assembly/pull/263/files#diff-9c2b84ac6a53e75567c56d4c450bb63bc246c5299418f044d56ad566ac044fc1, when upgrading karaf).

That system property was deprecated and effectively removed in karaf 4.2.2 (https://github.com/apache/karaf/pull/687/files#diff-1c660e549bdc0c28579f46d606d3aa36b18fe6b79fc7bfb45c03dd6f944949deR90). So the karaf admin role is no longer configurable and it is always `admin`.

I can't find who and when initially set the `karaf.admin.role` to `karaf_admin` in the server assembly (all the others assemblies kept the default `admin`), but I guess the intention was not to conflict with any Pentaho user or role also called `admin`.

**Short story**, since the upgrade to 4.2.6 the karaf's admin role is always `admin`; the `SpringSecurityLoginModule.KARAF_ADMIN` wasn't updated accordingly, so the user gets assigned a meaningless role `karaf_admin` and ends up with no admin rights.

Simply changing `SpringSecurityLoginModule.KARAF_ADMIN` to `admin` might do the trick. However, assigning extra roles like the Pull Request suggests might be fine or better.

Reply:
Thank you for the background. That answered by concerns.

I removed `SpringSecurityLoginModule.KARAF_ADMIN`. I'm going to stick with adding the 6 roles `"group,admin,manager,viewer,systembundles,ssh"`. As you can see "admin" is in there. I'm assuming the other roles are necessary and not covered under "admin" role.